Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonProducer.cc

```diff
@@ -0,0 +1,903 @@
+  desc.add<int>("useRaw", 0);
+  desc.add<bool>("collapseDepth", false);
+  desc.add<bool>("isItPlan1", false);
+  desc.addUntracked<int>("verbosity", 0);
+  desc.addUntracked<bool>("isItPreRecHit", false);
+  desc.addUntracked<bool>("writeRespCorr", false);
+  desc.addUntracked<std::string>("fileInCorr", "");
+  desc.addUntracked<int>("maxDepth", 4);
+  descriptions.add("alcaHcalHBHEMuonProducer", desc);
+}
+
+// ------------ method called once each job just after ending the event loop  ------------
+void AlCaHcalHBHEMuonProducer::endStream() {
+  globalCache()->nAll_ += nAll_;
+  globalCache()->nGood_ += nGood_;
+}
+
+void AlCaHcalHBHEMuonProducer::globalEndJob(const alcaHcalHBHEMuon::Counters* count) {
+  edm::LogVerbatim("HBHEMuon") << "Selects " << count->nGood_ << " out of " << count->nAll_ << " total # of events\n";
+}
+
+// ------------ method called when starting or ending a run  ------------
+void AlCaHcalHBHEMuonProducer::beginRun(edm::Run const& iRun, edm::EventSetup const& iSetup) {
+  hdc_ = &iSetup.getData(tok_ddrec_);
+  actHB.clear();
+  actHE.clear();
+  actHB = hdc_->getThickActive(0);
+  actHE = hdc_->getThickActive(1);
+#ifdef EDM_ML_DEBUG
+  unsigned int k1(0), k2(0);
+  edm::LogVerbatim("HBHEMuon") << actHB.size() << " Active Length for HB";
+  for (const auto& act : actHB) {
+    edm::LogVerbatim("HBHEMuon") << "[" << k1 << "] ieta " << act.ieta << " depth " << act.depth << " zside "
+                                 << act.zside << " type " << act.stype << " phi " << act.iphis.size() << ":"
+                                 << act.iphis[0] << " L " << act.thick;
+    HcalDetId hcid1(HcalBarrel, (act.ieta) * (act.zside), act.iphis[0], act.depth);
+    HcalDetId hcid2 = mergedDepth_ ? hdc_->mergedDepthDetId(hcid1) : hcid1;
+    edm::LogVerbatim("HBHEMuon") << hcid1 << " | " << hcid2 << " L " << activeLength(DetId(hcid2));
+    ++k1;
+  }
+  edm::LogVerbatim("HBHEMuon") << actHE.size() << " Active Length for HE";
+  for (const auto& act : actHE) {
+    edm::LogVerbatim("HBHEMuon") << "[" << k2 << "] ieta " << act.ieta << " depth " << act.depth << " zside "
+                                 << act.zside << " type " << act.stype << " phi " << act.iphis.size() << ":"
+                                 << act.iphis[0] << " L " << act.thick;
+    HcalDetId hcid1(HcalEndcap, (act.ieta) * (act.zside), act.iphis[0], act.depth);
+    HcalDetId hcid2 = mergedDepth_ ? hdc_->mergedDepthDetId(hcid1) : hcid1;
+    edm::LogVerbatim("HBHEMuon") << hcid1 << " | " << hcid2 << " L " << activeLength(DetId(hcid2));
+    ++k2;
+  }
+#endif
+
+  bool changed = true;
+  bool flag = hltConfig_.init(iRun, iSetup, processName_, changed);
+  edm::LogVerbatim("HBHEMuon") << "Run[" << nRun_ << "] " << iRun.run() << " hltconfig.init " << flag << std::endl;
+
+  theHBHETopology_ = &iSetup.getData(tok_htopo_);
+  const HcalRespCorrs* resp = &iSetup.getData(tok_respcorr_);
+  respCorrs_ = new HcalRespCorrs(*resp);
```

Comment:
`new` is discouraged, use `std::unique_ptr` instead

Reply:
taken care of